lib/trento.pm

```diff
@@ -1031,28 +1165,40 @@ sub cypress_exec {
     $args{timeout} //= bmwqemu::scale_timeout(600);
     my $ret = 0;
 
-    record_info('CY EXEC', "Cypress exec: $args{cmd}");
     my $image_name = CYPRESS_IMAGE . ":" . get_var('TRENTO_CYPRESS_VERSION', CYPRESS_DEFAULT_VERSION);
+    my $container_name = CYPRESS_IMAGE_TAG . random_string();
 
     # Container is executed with --name to simplify the log retrieve.
-    # To do so, we need to rm present container with the same name
-    script_run('podman images');
-    script_run('podman rm ' . CYPRESS_IMAGE_TAG . ' || echo "No ' . CYPRESS_IMAGE_TAG . ' to delete"');
-
-    my $cypress_entry_point = join(' ',
-        "'[",
-        '"/bin/sh",', '"-c",',
-        '"/usr/local/bin/cypress', $args{cmd}, '2>/results/cypress_' . $args{log_prefix} . '_log.txt"',
-        "]'");
-    my $cypress_run_cmd = join(' ', 'podman', 'run',
-        '-it', '--name', CYPRESS_IMAGE_TAG,
-        '-v', CYPRESS_LOG_DIR . ':/results',
-        '-v', "$args{cypress_test_dir}:/e2e", '-w', '/e2e',
-        '-e', '"DEBUG=cypress:*"',
-        "--entrypoint=$cypress_entry_point",
-        $image_name);
-    $ret = script_run($cypress_run_cmd, $args{timeout});
-    script_run('podman logs -t ' . CYPRESS_IMAGE_TAG) unless $ret;
+    # To avoid confusion, remove container from previous run
+    podman_delete_all();
+
+    my $cypress_log = CYPRESS_LOG_DIR . "/cypress_$args{log_prefix}_log.txt";
+    my $cypress_cmd = join(' ',
+        'podman', 'run',
+        '--name', $container_name,    # define a tag to retrieve the running container later
+        '-v', CYPRESS_LOG_DIR . ':/results',    # mount a folder to output results
+        '-v', "$args{cypress_test_dir}:/e2e",    # mount a folder to input the test code
+        '-w', '/e2e',
+        '-e "DEBUG=cypress:*"',
+        "--entrypoint cypress",    # doing so allow to specify more arguments for cypress later
+        $image_name,    # select the cypress image and its version
+        $args{cmd},    # the cypress operation to perform
+        '&>' . $cypress_log,    # redirect everything to file
+        '&'    # run podman in background
+    );
+    record_info('CY EXEC',
+        join("\n",
+            "container_name: $container_name",
+            "cmd:  $args{cmd}",
+            "cypress_cmd:  $cypress_cmd"));
+    enter_cmd('rm -rf ' . CYPRESS_LOG_DIR . '/*.*');
+    enter_cmd($cypress_cmd);
+    wait_serial('# ');
```

Comment:
`wait_serial()` will return the matched string on success or `undef` on timeout. In the case it returns `undef`, it means function did not find the prompt, which in turn can make commands executed on `podman_wait()` fail. As the first command on `podman_wait()` is a `script_output` which has `proceed_on_failure => 1`, wouldn't it be a good idea to try to handle such a situation here?

Reply:
Supposing all the Perl code here is fine, the intent is to have `$cypress_cmd` to run podman as a background process. So the time to wait here is the time to start the process and put it in background.